### PR TITLE
clean up errors in txn verifier / seq num checker

### PR DESCRIPTION
**Pull request by pattyshack**

### CODE REVIEW

File: fvm/transactionSequenceNum.go [hunk 1]
```diff
@@ -23,7 +23,12 @@ func (c *TransactionSequenceNumberChecker) Process(
 	txnState *state.TransactionState,
 	_ *programs.TransactionPrograms,
 ) error {
-	return c.checkAndIncrementSequenceNumber(proc, ctx, txnState)
+	err := c.checkAndIncrementSequenceNumber(proc, ctx, txnState)
+	if err != nil {
+		return fmt.Errorf("checking sequence number failed: %w", err)
```
Comment (pattyshack) [inline fvm/transactionSequenceNum.go:28]: note: previously, we weren't consistently wrapping errors with this msg.

File: fvm/transactionVerifier.go [hunk 2]
```diff
@@ -37,15 +37,12 @@ func (v *TransactionVerifier) Process(
...
 ) error {
-	return v.verifyTransaction(proc, ctx, txnState)
-}
-
-func newInvalidEnvelopeSignatureError(txSig flow.TransactionSignature, err error) error {
-	return errors.NewInvalidEnvelopeSignatureError(txSig.Address, txSig.KeyIndex, err)
-}
+	err := v.verifyTransaction(proc, ctx, txnState)
+	if err != nil {
+		return fmt.Errorf("transaction verification failed: %w", err)
```
Comment (pattyshack) [inline fvm/transactionVerifier.go:42]: note: previously, we weren't consistently wrapping errors with this msg.

File: fvm/transactionSequenceNum.go [hunk 3]
```diff
@@ -94,7 +98,7 @@ func (c *TransactionSequenceNumberChecker) checkAndIncrementSequenceNumber(
 		txnState.RunWithAllLimitsDisabled(func() {
 			_ = txnState.RestartNestedTransaction(nestedTxnId)
 		})
-		return fmt.Errorf("checking sequence number failed: %w", err)
+		return err
```
Comment (Tonix517) [inline fvm/transactionSequenceNum.go:101]: so we don't wrap `CodedError`?

Comment (pattyshack) [reply fvm/transactionSequenceNum.go:101]: the wrapping is now done in `Process` (line 28).  Previously, we weren't consistently wrapping the coded error (line 68, 75, 97 were wrap correctly, but not line 82)

**Review by Tonix517 — APPROVED** [review]

**Review by SaveTheRbtz — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**